internal/telemetry/client_test.go

```diff
@@ -61,14 +61,10 @@ func TestMetrics(t *testing.T) {
 	)
 	closed := make(chan struct{}, 1)
 
+	// we will try to set four metrics that the server must receive
+	expectedMetrics := 4
+
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if r.Header.Get("DD-Telemetry-Request-Type") == string(RequestTypeAppClosing) {
```

Comment:
We can't rely on app-closing to flush the metrics anymore, since app-closing is sent before metrics. Now we'll just hard code the expected metric messages and wait until we've hit that count.